packages/next/server/render.tsx

```diff
@@ -414,7 +414,7 @@ function createServerComponentRenderer(
   globalThis.__webpack_require__ = ComponentMod.__next_rsc__.__webpack_require__
   const Component = interopDefault(ComponentMod)
 
-  function ServerComponentWrapper({ App, router, ...props }: any) {
+  function ServerComponentWrapper({ router, ...props }: any) {
```

Comment:
I don't think if we should use the inherited `App` here. The case is even if there's `EnhancedApp` provided by the user, the server component still uses the default `App`.

(We should probably add a test for it.)

Reply:
If we're using RSC and user provides gIP on `Document`, it will fallback to builtin functional document to get rid of gIP. Then we won't have `ctx.renderPage` / `enhanceApp` thing for RSC. so we can only have App atm